hibernate-core/src/main/java/org/hibernate/engine/spi/NaturalIdResolutions.java

```diff
@@ -68,7 +68,7 @@ void manageSharedResolution(
 	/**
 	 * Removes any cross-reference from the L2 cache
 	 */
-	void removeSharedResolution(Object id, Object naturalId, EntityMappingType entityDescriptor);
+	void removeSharedResolution(Object id, Object naturalId, EntityMappingType entityDescriptor, boolean delayToAfterTransactionCompletion);
```

Comment:
Need to retain backwards compatibility since this is an SPI.

```suggestion
	void removeSharedResolution(Object id, Object naturalId, EntityMappingType entityDescriptor, boolean delayToAfterTransactionCompletion);
	
	default void removeSharedResolution(Object id, Object naturalId, EntityMappingType entityDescriptor, boolean delayToAfterTransactionCompletion) {
		removeSharedResolution( id, naturalId, entityDescriptor );
	}
```

Reply:
@beikov this your suggestion does not compile:
Duplicate method removeSharedResolution(Object, Object, EntityMappingType, boolean) in type NaturalIdResolutions